pelican/generators.py

```diff
@@ -560,57 +560,93 @@ def generate_tags(self, write):
         tag_template = self.get_template("tag")
         for tag, articles in self.tags.items():
             dates = [article for article in self.dates if article in articles]
-            write(
-                tag.save_as,
-                tag_template,
-                self.context,
-                tag=tag,
-                url=tag.url,
-                articles=articles,
-                dates=dates,
-                template_name="tag",
-                blog=True,
-                page_name=tag.page_name,
-                all_articles=self.articles,
-            )
+            try:
+                write(
+                    tag.save_as,
+                    tag_template,
+                    self.context,
+                    tag=tag,
+                    url=tag.url,
+                    articles=articles,
+                    dates=dates,
+                    template_name="tag",
+                    blog=True,
+                    page_name=tag.page_name,
+                    all_articles=self.articles,
+                )
+            except RuntimeError:
+                if not tag.slug:
+                    logger.info(
+                        'Tag "%s" has an invalid slug; skipping writing tag page...',
+                        tag,
+                        extra={"limit_msg": "Further tags with invalid slugs."},
+                    )
+                    continue
```

Comment:
With empty slugs already generating warnings, is this special case necessary anymore? Why not just do the `else` block?

Reply:
I think it's helpful to still keep this, because it makes it very explicity what Pelican is (or isn't doing), and it's not entirely obvious that an invalid slug would keep the page from being generated.

Also, it mirrors the logging for all the pages that *are* written.